csrc/scheduler/transpose.cpp

```diff
@@ -733,9 +729,14 @@ std::unique_ptr<TransposeParams> getTransposeHeuristics(
     // Adding a domain_guard so we can transform reference1
     ir_utils::TVDomainGuard domain_guard(reference1, cloned_1_td);
 
+    std::vector<int64_t> to_update(
+        {inner_most_pos1_in_ref1, inner_most_pos2_in_ref1});
     // we only apply split here, since we want to merge split dimensions, we can
     // simply map those merged domains via ContiguousInnerDimensionsMapper
-    scheduler_utils::splitDims(reference1, tparams->split_before_tiling);
+    scheduler_utils::splitDims(
+        reference1, tparams->split_before_tiling, to_update);
+    inner_most_pos1_in_ref1 = to_update[0];
```

Comment:
Is this change a general bug fix not specific to DID?

Reply:
I think it's a bug triggered by DID loop split. According to https://github.com/NVIDIA/Fuser/pull/3927#discussion_r1964824368, inner_most_pos1_in_ref1 and inner_most_pos2_in_ref2 ought to be loop axis but misused to access getLogicalDomain() in getVectorizationFactorTransposeGroup. Therefore, I also made this change: https://github.com/NVIDIA/Fuser/pull/3927/files#diff-364f51b47b1cf80c14b75d95ac2882238a65aa0594d588cff2fc54526307ca55L941

Reply:
I see. Looks like it was fine before because `logical == loop`. The change here makes sense.